mission/config/artillery.hpp

```diff
@@ -7,7 +7,7 @@ class vn_artillery_settings
         availability[] = {0, 1, 1, 0, 0};
         unit_trait_required = 1;
 	      // Distance from the edge of a blacklisted marker that a artillery/aircraft cannot be called in.
-	      danger_distance = 150;
+	      danger_distance = ["danger_close_distance", 150] call BIS_fnc_getParamValue;
```

Comment:
```suggestion
	      danger_distance = "['danger_close_distance', 150] call BIS_fnc_getParamValue";
```

Reply:
Is that needed because it's inside a config?
I tested it without quotations and it worked and didn't seem to get script errors in rpt log.

Reply:
The game preprocessor will auto-quote everything that's not a number, this should not be relied upon and explicit strings are preferred.

Reply:
Ahh alright. Thanks for the clarification!

Reply:
Config file is not parsing out the number from the parameters config it seems. I can only ever read back the sqf code as text.

@veteran29 with your suggested update (with quotes)

![image](https://github.com/user-attachments/assets/ecbc28ce-e1e7-41ed-bb47-0c9582eed616)

@Crowdedlight your original version (no quotes)

![image](https://github.com/user-attachments/assets/b66d9180-4bda-478c-87a2-f4dc8dd30a22)

Original version (no changes)

![image](https://github.com/user-attachments/assets/b463ed80-49f8-4dda-a447-aa66f0cba117)

----

Doing this in `para_server_init.sqf` does work, but also creates a confusing conflict between the configuration file and the server parameters.

```sqf
    // ...
    if (_airSupport != 0 || _artySupport != 0) then {

    // overwrite the config file setting on server initialisation
    missionNamespace setVariable [
        "vn_artillery_danger_distance",
        ['danger_close_distance', 5000] call BIS_fnc_getParamValue,
        true
    ];

    if (_airSupport == 0) then {
        vn_artillery_config_array set [0, nil];
    };
    // ...
```

Reply:
note -- this was tested on local player hosted, although i don't see how that should affect it (i'd assume arma loads config the same way for player and dedicated server)

Reply:
As I recall it worked fine on dedicated server back when I made the PR. At least I seem to remember setting the parameter small enough and be allowed to call in support on top of player positions. Maybe I did something wrong and the outcome just happend to allow the distances we wanted.

I am abroad at a conference atm, so can't test it until I am home again.

Reply:
hey @Crowdedlight you had a chance to check this out at all? no worries if not. it can wait for when you have time.

Reply:
I had a quick look. I think you are right. The reason we experienced it as working was because the ``getParam`` in the hpp likely failed and thus we just had 0, which meant we could call in firesupport right next to us, which was what we wanted. 

I failed successfully in the code changes 😅 

I think the better approach is to set it in the `para_server_init.sqf` based on the parameter as you said. Although not sure how that causes conflict between configuration file and server params? 

As I already see some params being set this way like: 
```sqf
// max_players_acav is classname of param
vn_mf_max_players_acav = ["max_players_acav", 99] call BIS_fnc_getParamValue;
publicVariable "vn_mf_max_players_acav";
```

Although I was looking for the code that uses the parameter values to see if the danger-close check exists player side or server side. But I can't seem to find the code responsible for the check or the execution of the RTO fire support. I seem to remember I found it back when I made the changes originally, but can't remember where 😅